app/code/core/Mage/Shipping/Model/Resource/Carrier/Tablerate.php

```diff
@@ -128,7 +128,7 @@ public function getRate(Mage_Shipping_Model_Rate_Request $request)
 
         $i = 0;
         $postcode = $request->getDestPostcode();
-        while (strlen($postcode) > 1) {
+        while (strlen($postcode ?? '') > 1) {
```

Comment:
```suggestion
        $postcode = (string)$request->getDestPostcode();
        while (strlen($postcode) > 1) {
```
It's more efficient as it's outside the loop.

Reply:
good catch @ADDISON74, I like @kiatng version, it looks neater to me because it doesn't get executed inside the loop